### PR TITLE
Fix Vultr cloud_interfaces usage

### PR DESCRIPTION
**Pull request by eb3095**
## Proposed Commit Message
```
cloud_interfaces is intended to be a netcfg. Change the helper function to reflect this.
```

## Checklist:

 - [X] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [X] I have updated or added any unit tests accordingly
 - [X] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/sources/helpers/vultr.py [hunk 1]
```diff
@@ -276,8 +276,10 @@ def generate_interface_additional_addresses(interface, netcfg):
 
 
 # Make required adjustments to the network configs provided
-def add_interface_names(interfaces):
-    for interface in interfaces:
+def add_interface_names(netcfg):
+    for interface in netcfg["config"]:
```
Comment (aciba90) [inline cloudinit/sources/helpers/vultr.py:280]: This change implies that either:

1. This function has never been functional, or
2. There was a change in the instance metadata schema under `cloud_interfaces`. See https://github.com/canonical/cloud-init/blob/aacdff5b9298db2acbb5d10f0a42cb06d2c0a745/cloudinit/sources/DataSourceVultr.py#L73

**Review by aciba90 — COMMENTED** [review]
Thanks, @eb3095.

I have been trying to look for [documentation](https://www.vultr.com/metadata/) about the `cloud_interfaces` key without success. Could you please help us to understand why this change is required?

Additionally, this is not covered by any unit tests. Could we please minimally cover it?

File: cloudinit/sources/helpers/vultr.py (hunk 1 above)
Comment (eb3095) [reply cloudinit/sources/helpers/vultr.py:280]: Its never been functional nor has it ever been used. This is our way of dropping in a fully configured, outside of interface names, cloud-init netcfg file in. Its not currently documented anywhere. As for unit tests, I will add that now.

Comment (aciba90) [reply cloudinit/sources/helpers/vultr.py:280]: Thanks for the context and the unit test.

**Comment by eb3095** [comment]
@aciba90 fixed the unit test and got this passing. Sorry ran into some python weirdness with the dict I had to work around. Looks to be acting correctly now however.

**Review by aciba90 — APPROVED** [review]
LGTM. Thanks, @eb3095.

**Pull request merged**